### PR TITLE
caeml: init at unstable-2023-05-24

### LINKED ISSUE #236909: Package request: caeml
**Project description**

This project reads email from stdin or input files and outputs a more human readeable format to stdout, it is usefull for two uses:
a) Poking in your maildir folder when trying to figure out what's wrong with your syncing.
b) previewing `message/rfc822` attachments in [aerc](https://github.com/ferdinandyb/caeml/blob/master/aerc-mail.org)

**Metadata**

* homepage URL: https://github.com/ferdinandyb/caeml
* source URL: https://github.com/ferdinandyb/caeml
* license: MIT
* platforms: I guess everything under the sun where go's supported?

### PR DESCRIPTION
**Pull request by emilytrau**
###### Description of changes

Resolves #236909 @Juliaria08

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/mailreaders/caeml/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+, buildGoModule
+, fetchFromGitHub
+}:
+buildGoModule rec {
+  pname = "caeml";
+  version = "unstable-2023-05-24";
+
+  src = fetchFromGitHub {
+    owner = "ferdinandyb";
+    repo = pname;
```
Comment (Janik-Haag) [inline pkgs/applications/networking/mailreaders/caeml/default.nix:11]: nitpick: 
```suggestion
    repo = finalAttrs.pname;
```
and remove rec

**Review by Janik-Haag — APPROVED** [review]
LGTM; you might want to change the nitpick

File: pkgs/applications/networking/mailreaders/caeml/default.nix (hunk 1 above)
Comment (emilytrau) [reply pkgs/applications/networking/mailreaders/caeml/default.nix:11]: looks like `buildGoModule` doesn't support the `finalAttrs` pattern yet :/

Comment (Janik-Haag) [reply pkgs/applications/networking/mailreaders/caeml/default.nix:11]: Yeah sorry, didn't know that :see_no_evil:

**Pull request merged**